### ISSUE TITLE
Array of byte not finding Address

### ISSUE THREAD

**Issue by Karlos007** [cheat-engine/cheat-engine#1316]
Unless I specify the region...
![aob](https://user-images.githubusercontent.com/23475289/90269101-6028db00-de58-11ea-8b83-62a1216e07fa.png)
Maybe the process (yuzu) is to big in memory? 
Also lua / asm aobscans don't find either.

Cheat Engine 7.1 (Also tried 7.0 and different settings / debuggers)
Windows 10 pro v2004 64 bits
32gb ram / i7-6700 cpu

**Comment by mgrinzPlayer** [comment]
Is it related with this commit? c0534e3777d2d78a95e4b6d0b81de79e05c8866e
Did you try to compile CE from source and above issue still exists?

**Comment by Karlos007** [comment]
I'm trying to compile the 64 bit but only the 32 app compiles :/ and that doesn't scan at all.

lazarus-2.0.10-fpc-3.2.0-win64
lazarus-2.0.10-fpc-3.2.0-win32

lazarus-2.0.10-fpc-3.2.0-cross-x86_64-win64-win32
lazarus-2.0.10-fpc-3.2.0-cross-i386-win32-win64

What files should I install?

**Comment by Karlos007** [comment]
![aob1](https://user-images.githubusercontent.com/23475289/90278225-59559480-de67-11ea-839f-708b930870ab.png)

**Comment by mgrinzPlayer** [comment]
Use current source files - use Git client and clone this repository or download them from this link: https://github.com/cheat-engine/cheat-engine/archive/master.zip

**Comment by Karlos007** [comment]
Thanks that link worked, I was using https://github.com/cheat-engine/cheat-engine/archive/7.1.zip
Still doesn't find anythink unless I narrow the Memory Scan

**Comment by mgrinzPlayer** [comment]
Does it find when you tick MEM_MAPPED option in "CE Settings->Scan settings"?

If not, you can try modify memscan.pas file, something like changing the lines with "size: dword" or "size: integer" to "size: qword"...

**Comment by Karlos007** [comment]
- yeah MEM_MAPPED is ticked 
- Still the same changing line 5232:  size: dword ->  size: qword
- It's weird because other scans work if they are early in memory (this one is 212629A0A50)
Start 0000000000000000 Stop 7fffffffffffffff doesn't find
Start          21262900000  Stop 7fffffffffffffff does find...

**Comment by mgrinzPlayer** [comment]
I wrote "something like ...".

You can try other places too. Maybe: 
- "addressSize: dword" => "addressSize: qword" 
- "memorySize: dword" => "memorySize: qword"
- "maxregionsize: dword" => "maxregionsize: dword"

If it still doesn't work, then, I think there's no an easy fix like this one c0534e3777d2d78a95e4b6d0b81de79e05c8866e

**Comment by Karlos007** [comment]
I don't know why but that worked, seems to be related to that 2GB, even if it's not unknown initial value.
Also makes the lua scripts work.
Thank you so much for your help, I'm gonna keep making my cheat table :)

**Comment by mgrinzPlayer** [comment]
You can help and find the smallest possible change which fixes your issue.

Extract source files again. Then:
- do one change, compile
- check
- if problem exists, do next change, compile
- check


Note: while doing this, you should use Lazarus "Run=>Clean up and build project" (all options selected except "package output directories").

**Comment by Karlos007** [comment]
Seems that 
456: maxregionsize: qword; does the job
Normal Cheat Engine 7 doesn't find anything, but extracting source files and changing that line finds the address.

**Comment by mgrinzPlayer** [comment]
OK. Thank you. This information should be useful. I'm adding permalink to this line:
https://github.com/cheat-engine/cheat-engine/blob/5850cd01f635240ffdf51b9e62ac9d0ff96f2b7f/Cheat%20Engine/memscan.pas#L456

**Comment by cheat-engine** [comment]
what is your buffersize in scansettings?

**Comment by Karlos007** [comment]
1024 KB Currently working fine with my compiled CE.
I was changing between 512 and 1024 and more settings but always failed to scan the one that I not compiled.

**Comment by cheat-engine** [comment]
i can't see why maxregionsize would cause a difference as it's size is limited by buffersize.
but i'll check (tomorrow)

**Comment by mgrinzPlayer** [comment]
@cheat-engine maybe those lines:
https://github.com/cheat-engine/cheat-engine/blob/5850cd01f635240ffdf51b9e62ac9d0ff96f2b7f/Cheat%20Engine/memscan.pas#L6089
https://github.com/cheat-engine/cheat-engine/blob/5850cd01f635240ffdf51b9e62ac9d0ff96f2b7f/Cheat%20Engine/memscan.pas#L6560

currentblocksize type is qword.

**Comment by Karlos007** [comment]
I also have to sleep :)
![aob2](https://user-images.githubusercontent.com/23475289/90297501-fed02e80-de8e-11ea-8757-be44cdea9a87.png)
I deleted the entire folder cheat-engine-master and unziped again before editing that one line.
I used "Run=>Clean up and build project"; now has 1130 files but when I tried before it was 0 since I deleted all.

**Comment by cheat-engine** [comment]
How many bytes is the AOB you're scanning for ? more than 2000?

**Comment by Karlos007** [comment]
The array is around 50.
I have others with 500 that work fine.
Complied in debug-nomt 64-bit with the original master folder? Without touching line 456?

**Comment by mgrinzPlayer** [comment]
>Without touching line 456?

Do both, the more information the better.

**Comment by Karlos007** [comment]
Both find the address, but takes **much more** time scanning. Around 3 minutes instead of 15 sec.
I see that master memscan.pas 456 it's now maxregionsize: qword; instead of maxregionsize: dword;

**Comment by mgrinzPlayer** [comment]
To be perfectly sure. You extracted source files from that zip file you downloaded earlier, right?

To summarize it. This?
| | line 456 has dword | line 456 has qword |
|--|--|--|
| build mode is `release 64bit` | doesn't find | finds |
| build mode is `debug-nomt 64-bit` | finds/slow | finds/slow |

<p> </p>

or this?
| | line 456 has dword | line 456 has qword |
|--|--|--|
| build mode is `release 64bit` | doesn't find | finds |
| build mode is `debug-nomt 64-bit` | doesn't find | finds/slow |

**Comment by Karlos007** [comment]
Yep the first one, debug-nomt 64-bit slowly finds with dword and qword
release finds with qword only
(7.1 / 7.0 / 6.8.3 .exe from the github page don't find either)
